src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Layout/AnchorLayoutTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Drawing;
+using System.Windows.Forms.Primitives;
+
+namespace System.Windows.Forms.Layout.Tests
+{
+    public class AnchorLayoutTests : IClassFixture<ThreadExceptionFixture>
+    {
+        private static (Form, Button) GetFormWithAnchoredButton()
+        {
+            Form form = new ();
+            form.Size = new Size(200, 300);
+            Button button = new ();
+            button.Location = new Point(20, 30);
+            button.Size = new Size(20, 30);
+            button.Anchor = AnchorStyles.Left | AnchorStyles.Top | AnchorStyles.Right | AnchorStyles.Bottom;
+
+            return (form, button);
+        }
+
+        private static void Dispose(Form form, Button button)
+        {
+            button?.Dispose();
+            form?.Dispose();
+        }
+
+        [WinFormsFact]
+        public void ControlNotParented_NoAnchorsComputed()
+        {
+            var (form, button) = GetFormWithAnchoredButton();
+            DefaultLayout.AnchorInfo anchorInfo = DefaultLayout.GetAnchorInfo(button);
+            Assert.Null(anchorInfo);
+            _ = button.Handle; // Force handle creation;
```

Comment:
We don't use this pattern, instead we'be using either 
```suggestion
            button.CreateControl();
```
\- or -
```suggestion
           Assert.NotEqual(IntPtr.Zero, button.Handle);
```

Reply:
Create control does lot more. I just wanted to create Handle.

Reply:
Could you please add a comment to the test explaining the scenario where Handle is created but control is not.

Also, I prefer an Assert to a comment, but this is up to you.

Reply:
> Create control does lot more. I just wanted to create Handle.

Then please use the 2nd snippet.
I've looked at the `CreateControl` code and I don't really see "lot more work" there in this scenario besides settings flags. If you could provide some insights and, and how that affects the tests it would be great.

Reply:
In this scenario functionally you may not see difference but i don't need to call that extra code from `CreateControl` (which sets some flag and raise event) method. Test was to ensure Handle creation doesn't alter Anchors yet. Well, if you both feel Assert is more meaningful, i will change to it but by then we are hiding the fact that Handle is just created.

Reply:
> by then we are hiding the fact that Handle is just created.

I don't think we are hiding anything here. Accessing `Handle` force-creates a handle, if it's not created. 
This has been the source of many bugs.